### DISCUSSION TITLE
Searching for meeting notes

### DISCUSSION THREAD

**Discussion by kaibagley** [alphapapa/org-ql#457] [Q&A] [answered]
Hey guys,

Sorry if this is obvious/been answered elsewhere but I couldn't find anything

I want to search my todo list for old research meeting notes. This would be done along the lines of `(and (todo "MEET") (tags "research"))`. However I'd like to also narrow it down to the attendees, or some other regexp within the body of the MEET subtree. Here is an example MEET:

```
** MEET Research Meeting                                  :meeting:research:
*** Attendees
- [X] Steve
- [X] John

*** To Discuss
- Stuff to discuss
- Topic A

*** Tasks
- Task A
- Task B
```
I've tried `(and (todo "MEET") (tags "research") (regexp/rifle "John"))`, but this will only return entries where "John" is in the actual headline rather than the body.

Is there some other query type I'm missing or is this something that isn't currently possible?

Thanks for the great software!!

**Comment by alphapapa** [comment, marked as answer]
`regexp/rifle` is not a valid predicate.  As the documentation explains, `regexp` "Matches against entire entry, from beginning of its heading to the next heading."  In your example, the text `John` is not present in an entry having a `MEET` to-do keyword.  You probably want to use the `parent` or `ancestor` predicate for the keyword, e.g. `(and (ancestor (todo "MEET")) (tags "research") "John")`.

**Reply by kaibagley** [reply]
Perfect, thanks a lot! By `regexp/rifle` I meant that I had tried them both, but I see now why neither would for my purpose work due to the `todo "MEET"` line.
